sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/TableFileCatalog.scala

```diff
@@ -42,24 +43,21 @@ class TableFileCatalog(
 
   protected val hadoopConf = sparkSession.sessionState.newHadoopConf
 
+  private val fileStatusCache = FileStatusCache.getOrInitializeShared(new Object(), sparkSession)
```

Comment:
Odd, I figured we'd call the `refresh` method to refresh the table.

The reason I suggested using the table's name is so that all references to the same table can use the same cache—avoid redundant cache entries and provide the ability to inherit cached entries for new instances of the same logical table. Aside from implementation issues, does that make sense?

Anyway, this is something we can explore in a follow up PR if desired.